### PR TITLE
fix(template-compiler): make duplicate slots an error, not warning

### PR DESCRIPTION
**Pull request by nolanlawson**
## Does this pull request introduce a breaking change?

* 🚨 Yes, it does introduce a breaking change.

Duplicate slots, or slots inside of an iterator, will now throw an error instead of warning.

## Does this pull request introduce an observable change?

* ⚠️ Yes, it does include an observable change.

## GUS work item
W-10930657
W-10930770

### CODE REVIEW

File: packages/@lwc/template-compiler/src/__tests__/fixtures/parsing-errors/duplicate-default-slot-light/actual.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<template lwc:render-mode="light">
+    <slot></slot>
+    <slot></slot>
+</template>
```
Comment (nolanlawson) [inline packages/@lwc/template-compiler/src/__tests__/fixtures/parsing-errors/duplicate-default-slot-light/actual.html:4]: I also added some tests for light DOM mode to ensure that light DOM has the same behavior as shadow DOM.

**Comment by nolanlawson** [comment]
Reason for `nomerge` label: we should do an analysis of the blast radius for introducing this change first.

**Review by pmdartus — APPROVED** [review]

**Comment by nolanlawson** [comment]
We can't actually do this; it'd be too breaking.

**Pull request closed without merging**